modules/azure/api_connectors/service_bus_managed_identity/connection.json

```diff
@@ -1,29 +1,22 @@
 {
   "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
   "contentVersion": "1.0.0.0",
-  "metadata": {
-    "_generator": {
-      "name": "bicep",
-      "version": "0.9.1.41621",
-      "templateHash": "14233748139416635108"
-    }
-  },
   "parameters": {
     "location": {
-      "type": "string",
+      "type": "String",
       "defaultValue": "[resourceGroup().location]",
```

Comment:
Why is there still a default value here when the location variable in terraform already has a default?

Reply:
Good point indeed, I will set it to use `local.globals.location`

Reply:
I misunderstood, in the terraform module, there is a location parameter which is not in the ARM template. This is in Terraform itself:

`
template_content = file("./connection.json") **<< ARM**
  parameters_content = jsonencode({
    "service_bus_connection_name" = {
      value = var.connection_name
    }
    "service_bus_namespace_endpoint" = {
      value = var.service_bus_namespace_endpoint
    }
    "location" = {
      value = location **<< Other Location **
    }
`

Reply:
Why not just remove the default, because the variable wil always be filled with a value due to its own default, this default will never be used.

Reply:
I checked and that works fine too. I changed it.

Reply:
Tested with and without location var used?

Reply:
Without the default, it will be set from globals. But this is a module, we need to be able to change it if we want besides the global variable. If I remove the declaration of the location, Terraform sees differences, so that is needed, and it cannot be null or empty as well > difference.

Reply:
This means we can change it if we want, but it uses the default as well. So backwards compatible too.